src/main/java/org/takes/rq/multipart/RqMtFake.java

```diff
@@ -50,120 +59,133 @@
-                    .append(body)
-                    .append(RqMtFake.CRLF);
-            }
+    private static final class FakeMultipartRequest extends RqWrap {
+        /**
+         * Ctor.
+         * @param rqst The Request object
+         * @param list The sequence of dispositions
+         * @throws IOException if can't process requests
+         */
+        FakeMultipartRequest(final Request rqst, final Request... list)
+            throws IOException {
+            this(rqst, new RqMtFake.FakeBody(list));
+        }
+
+        /**
+         * Ctor.
+         * @param rqst The Request object
+         * @param body The body of dispositions
+         * @throws IOException if can't process requests
+         */
+        FakeMultipartRequest(final Request rqst, final Body body)
+            throws IOException {
+            super(
+                new RequestOf(
+                    new RqWithHeaders(
+                        rqst,
+                        String.format(
+                            "Content-Type: multipart/form-data; boundary=%s",
+                            RqMtFake.BOUNDARY
+                        ),
+                        String.format(
+                            "Content-Length: %s",
+                            new LengthOf(new InputOf(body.body())).intValue()
+                        )
+                    ),
+                    body
+                )
+            );
         }
-        builder.append("Content-Transfer-Encoding: utf-8")
-            .append(RqMtFake.CRLF)
-            .append(String.format("--%s--", RqMtFake.BOUNDARY));
-        return builder;
     }
 
     /**
-     * This class is using a decorator pattern for representing
-     * a fake HTTP multipart request.
+     * Fake body .
      * @since 0.33
      */
-    private static final class FakeMultipartRequest implements Request {
-        /**
-         * Request object. Holds a value for the header.
-         */
-        private final Request req;
+    private static final class FakeBody implements Body {
 
         /**
-         * Holding multiple request body parts.
+         * Content of the body.
          */
-        private final String parts;
+        private final String content;
 
         /**
-         * The Constructor for the class.
-         * @param rqst The Request object
-         * @param list The sequence of dispositions
-         * @throws IOException if can't process requests
+         * Ctor.
+         * @param parts Fake request body partsFakeBody
+         * @throws IOException If fails
          */
-        FakeMultipartRequest(final Request rqst, final Request... list)
-            throws IOException {
-            this.req = rqst;
-            this.parts = RqMtFake.fakeBody(list).toString();
+        @SuppressWarnings(
+            "PMD.ConstructorOnlyInitializesOrCallOtherConstructors"
+        )
+        private FakeBody(final Request... parts) throws IOException {
```

Comment:
@andreoss Please remove this code from constructor and the warning related

Reply:
@paulodamaso Fixed.